backend/rest/rest.go

```diff
@@ -0,0 +1,212 @@
+package rest
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/restic/restic/backend"
+)
+
+const connLimit = 10
+
+// Returns the url of the resource
+func restPath(url *url.URL, t backend.Type, name string) string {
+	location := url.String()
+	if !strings.HasSuffix(location, "/") {
+		location += "/"
+	}
+	if t == backend.Config {
+		return location + string(t)
+	}
+	return location + string(t) + "/" + name
+}
+
+type RestBlob struct {
+	b     *Rest
+	buf   *bytes.Buffer
+	final bool
+}
+
+func (rb *RestBlob) Write(p []byte) (int, error) {
+	if rb.final {
+		return 0, errors.New("blob already closed")
+	}
+	n, err := rb.buf.Write(p)
+	return n, err
+}
+
+func (rb *RestBlob) Read(p []byte) (int, error) {
+	return rb.buf.Read(p)
+}
+
+func (rb *RestBlob) Close() error {
+	rb.final = true
+	rb.buf.Reset()
+	return nil
+}
+
+// Finalize moves the data blob to the final location for type and name.
+func (rb *RestBlob) Finalize(t backend.Type, name string) error {
+	if rb.final {
+		return errors.New("Already finalized")
+	}
+
+	rb.final = true
+
+	// Check key does not already exist
+	resp, err := http.Get(restPath(rb.b.url, t, name))
```

Comment:
I think that this is not a good idea. If the the blob at that location is 8MiB, wouldn't that download the whole blob? Instead, maybe `HEAD` is better? Or skip this check and `POST` the new content, and expect an error from the server when the resource already exists (this expected behaviour must be documented)? REST uses `PUT` to overwrite an already-existing resource...